### PR TITLE
Resolve `test_stats_aggregation`

### PR DESCRIPTION
**Pull request by sarahyurick**
Currently working on `test_stats_aggregation()` in `test_groupby.py`... so far have `REGR_COUNT` functionality and working on others.

### CODE REVIEW

File: dask_sql/physical/rel/logical/aggregate.py [hunk 1]
```diff
@@ -378,8 +378,12 @@ def _collect_aggregations(
                     df = df.assign(**{two_columns_proxy: (~is_null(df[col1]))})
```
Comment (sarahyurick) [inline dask_sql/physical/rel/logical/aggregate.py:378]: I haven't checked what happens if we go into this `if` block yet.

File: dask_sql/physical/rel/logical/aggregate.py [hunk 2]
```diff
@@ -157,6 +157,48 @@ class DaskAggregatePlugin(BaseRelPlugin):
...
+                    sum_of_squares.sum(),
+                ),
+                lambda count, sum, sum_of_squares: (
+                    sum_of_squares - (sum * (sum / count))
+                ),
+            )
+        ),
+        "regr_sxx": AggregationSpecification(
+            dd.Aggregation(
+                "regr_sxx",
```
Comment (sarahyurick) [inline dask_sql/physical/rel/logical/aggregate.py:176]: `regr_syy` and `regr_sxx` currently just output rows with 0.0, which I suspect has something to do with the code here not properly handling two inputs. From [this site](https://docs.teradata.com/r/kmuOwjp1zEYg98JsB8fu_A/ZsXbiMrja5EYTft42VoiTQ), it looks like I only need the first parameter of the function, but I'm having trouble formatting the aggregation with `dd.Aggregation`

File: dask_sql/physical/rel/logical/aggregate.py [hunk 3]
```diff
@@ -367,8 +409,8 @@ def _collect_aggregations(
             # Gather information about the input column
             inputs = agg.getArgs(expr)
 
-            # TODO: This if statement is likely no longer needed but left here for the time being just in case
-            if aggregation_name == "regr_count":
+            # if aggregation_name in ["regr_count", "regr_syy", "regr_sxx"]:
+            if len(inputs) == 2:
```
Comment (sarahyurick) [inline dask_sql/physical/rel/logical/aggregate.py:413]: Looks like this code was initially designed where `regr_count` is the only aggregation we expected to have 2 inputs. I changed it to `if len(inputs) == 2` instead, although I'm not sure that's what we want with the null logic used for `regr_count` that is included in the code block below.

Maybe `if aggregation_name in ["regr_count", "regr_syy", "regr_sxx"]` would be better?

File: tests/integration/test_groupby.py [hunk 4]
```diff
@@ -357,122 +357,119 @@ def test_stddev(c, gpu):
...
-    reason="WIP DataFusion - https://github.com/dask-contrib/dask-sql/issues/463"
-)
 def test_stats_aggregation(c, timeseries_df):
     # test regr_count
     regr_count = c.sql(
         """
     SELECT
         name,
-        COUNT(x) FILTER (WHERE y IS NOT NULL) AS expected,
+        -- COUNT(x) FILTER (WHERE y IS NOT NULL) AS expected,
```
[the remark this answers is not in the document]
Comment (sarahyurick) [reply tests/integration/test_groupby.py:366]: Depends on #760

File: tests/integration/test_groupby.py [hunk 5]
```diff
@@ -417,59 +414,60 @@ def test_stats_aggregation(c, timeseries_df):
...
+    # GROUP BY ts.name
+    # """
+    # ).fillna(0)
+
+    # assert_eq(
+    #     covar_pop["expected"],
+    #     covar_pop["calculated"],
+    #     check_dtype=False,
+    #     check_names=False,
+    # )
```
Comment (charlesbluca) [inline tests/integration/test_groupby.py:443]: Can we uncomment these tests now?

File: tests/integration/test_groupby.py [hunk 6]
```diff
@@ -417,59 +414,60 @@ def test_stats_aggregation(c, timeseries_df):
...
+    # GROUP BY ts.name
+    # """
+    # ).fillna(0)
+
+    # assert_eq(
+    #     covar_samp["expected"],
+    #     covar_samp["calculated"],
+    #     check_dtype=False,
+    #     check_names=False,
+    # )
```
Comment (charlesbluca) [inline tests/integration/test_groupby.py:471]: This one as well

**Review by charlesbluca — COMMENTED** [review]
I imagine these tests will fail on GPU due to the issues described in #681, so might want to xfail on GPU for now

**Comment by sarahyurick** [comment]
> I imagine these tests will fail on GPU due to the issues described in #681, so might want to xfail on GPU for now

Sure, do you mean to add a `@pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])`?

File: tests/integration/test_groupby.py (hunk 5 above)
Comment (sarahyurick) [reply tests/integration/test_groupby.py:443]: I was planning to resolve `covar_pop` and `covar_samp` in a different PR, and have opened issue #753 to track it.

**Comment by ayushdg** [comment]
> Sure, do you mean to add a @pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])?

Maybe split them out into multiple tests mark the covar one to xfail and the regr ones to xfail only on gpu

**Comment by sarahyurick** [comment]
> Maybe split them out into multiple tests mark the covar one to xfail and the regr ones to xfail only on gpu

Sounds good, thanks @ayushdg and @charlesbluca !

File: tests/integration/test_groupby.py [hunk 7]
```diff
@@ -354,10 +354,11 @@ def test_stddev(c, gpu):
     c.drop_table("df")
 
 
-@pytest.mark.skip(
-    reason="WIP DataFusion - https://github.com/dask-contrib/dask-sql/issues/463"
-)
-def test_stats_aggregation(c, timeseries_df):
+@pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])
```
Comment (ayushdg) [inline tests/integration/test_groupby.py:357]:
```suggestion
@pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=(pytest.mark.gpu, pytest.mark.skip))])
```

**Review by charlesbluca — APPROVED** [review]

**Pull request merged**